server/src/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -747,6 +810,67 @@ public VpcOffering updateVpcOffering(final long vpcOffId, final String vpcOfferi
     }
 
     @Override
+    public CIDR[] getZoneSuperCidrList(final long zoneId) throws BadCIDRException {
+        final String superCIDRList = _dcDao.getDetail(zoneId, Params.SUPER_CIDR.name());
+        if (superCIDRList != null) {
+            String[] str_cidr_list = superCIDRList.split(",");
+            CIDR[] cidr_list = NetUtils.convertToCIDR(str_cidr_list);
+            if (!NetUtils.cidrListConsistency(cidr_list)) {
+                throw new InvalidParameterValueException("The cidr list is not consistent " + Arrays.toString(cidr_list));
+            }
```

Comment:
As noted in earlier comments, please consider a CIDR custom collection type and encapsulation of this logic in it.